DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.cpp

```diff
@@ -0,0 +1,260 @@
+//////////////////////////
+// **** Board Test **** //
+//////////////////////////
+
+// Purpose of this test: To check board interfaces
+// and some of the oneAPI shim metrics like
+// host-to-device interface bandwidth,
+// Kernel clock frequency
+// Kernel launch latency
+// Kernel-to-memory bandwidth
+// The complete test comprises of 6 tests listed below
+// User can choose to run all or select a specific test using options 0-6
+// Usage: ./board_test.fpga to run all tests
+// ./board_test.fpga -test=<test-number> to run a specific test
+// ./board_test.fpga -help to print details about these tests
+// Tests:
+// 0 = Default (Run all tests)
+// 1 = Host Speed Test
+// 2 = Kernel Clock Frequency Test
+// 3 = Kernel Launch Test
+// 4 = Kernel Latency Measurement
+// 5 = Kernel-to-Memory Read Write Test
+// 6 = Kernel-to-Memory Bandwidth Test
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+#include <iostream>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities/<version>/include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+// Test related header files
+#include "board_test.hpp"
+
+using namespace sycl;
+
+// Global variable used in kernel_clk_freq.cpp and kernel_mem_bw.cpp test
+// float bandwidth_nop = 0.0;
+
+int main(int argc, char* argv[]) {
+  // Always print small help at the beginning of board test
+  PrintHelp(0);
+
+  // Default is to run all tests
+  int test_to_run = 0;
+
+  // test_to_run value changed according to user selection if user has provided
+  // an input using "-test=<test_number>" option (see PrintHelp function for
+  // test details)
+  if (argc == 2) {
+    std::string tmp_test(argv[1]);
+    if ((tmp_test.compare(0, 6, "-test=")) == 0) {
+      test_to_run = std::stoi(tmp_test.substr(6));
+      if (test_to_run < 0 || test_to_run > 7) {
+        std::cout
+            << "Not a valid test number, please select the correct test from "
+            << "list above and re-run binary with updated value.\n\n";
+        return 1;
+      }
+    } else if ((tmp_test.compare(0, 5, "-help")) == 0) {
+      PrintHelp(1);
+      return 0;
+    } else {
+      std::cout << "Incorrect argument passed to ./board_test.fpga! Cannot run "
+                << "test\n, please refer to usage information above for "
+                << "correct command.\nTerminating test!\n";
```

Comment:
```suggestion
                << "correct command."
                << std::endl;
                << "Terminating test!" 
                << std::endl;
```
Please remove all uses of `\n` and replace them with `std::endl`

Reply:
current guidelines suggest \n, keeping \n as per discussion. Let me know if this needs to be changed in future.